src/tests/lvm/lvm_shared_mount/lvm_shared_mount_test.go

```diff
@@ -0,0 +1,265 @@
+package lvm_shared_mount_volume
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openebs/openebs-e2e/common"
+	"github.com/openebs/openebs-e2e/common/e2e_agent"
+	"github.com/openebs/openebs-e2e/common/e2e_ginkgo"
+	"github.com/openebs/openebs-e2e/common/k8stest"
+	"github.com/openebs/openebs-e2e/common/lvm"
+	coreV1 "k8s.io/api/core/v1"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Background:
+//     Given a k8s cluster is running with the product installed
+// Scenario: shared mount volume creation for lvm volume
+//     Given a volume has been successfully created with shared mount enabled via storage class
+//     When two applications are deployed using the same volume
+//     Then both applications should be in running state using the same volume
+
+var nodeConfig lvm.LvmNodesDevicePvVgConfig
+
+func setupLVM() {
+	loopDevice := e2e_agent.LoopDevice{
+		Size:   10737418240,
+		ImgDir: "/tmp",
+	}
+
+	workerNodes, err := lvm.ListLvmNode(common.NSOpenEBS())
+	Expect(err).ToNot(HaveOccurred(), "failed to list worker nodes")
+
+	nodeConfig = lvm.LvmNodesDevicePvVgConfig{
+		VgName:        "lvmvg",
+		NodeDeviceMap: make(map[string]e2e_agent.LoopDevice),
+	}
+	for _, node := range workerNodes {
+		nodeConfig.NodeDeviceMap[node] = loopDevice
+	}
+
+	logf.Log.Info("Setting up node with loop device, PV, and VG", "node config", nodeConfig)
+	err = nodeConfig.ConfigureLvmNodesWithDeviceAndVg()
+	Expect(err).ToNot(HaveOccurred(), "failed to setup node")
+}
+
+func volumeSharedMountTest(decor string, engine common.OpenEbsEngine, volType common.VolumeType, fstype common.FileSystemType, volBindModeWait bool) {
```

Comment:
As it stands this function cannot be used for raw block volumes.
That's okay, but remove the volType argument from the formal parameters.
For raw block the commands will be quite different so would require a different function.
If the intent is to add check for raw block volumes then I'd rename this function to something like
`fsVolumeSharedMountTest`

Reply:
Updated!!